src_new/controllers/uas.js

```diff
@@ -0,0 +1,441 @@
+var CONFIG = require("../config.js");
+var CONSTANTS = require("../constants.js");
+var UTIL = require("../util.js");
+var adapterManager = require("../adapterManager.js");
+var bidManager = require("../bidManager.js");
+var SLOT = require("../slot.js");
+var PHOENIX = require("uas-adclient");
+
+var wrapperTargetingKeys = {}; // key is div id
+/* start-test-block */
+exports.wrapperTargetingKeys = wrapperTargetingKeys;
+/* end-test-block */
+var slotSizeMapping = {}; // key is div id
+/* start-test-block */
+exports.slotSizeMapping = slotSizeMapping;
+/* end-test-block */
+var slotsMap = {}; // key is div id, stores the mapping of divID ==> googletag.slot
+
+/* start-test-block */
+exports.slotsMap = slotsMap;
+/* end-test-block */
+
+var windowReference = null;
+var refThis = this;
+
+function setWindowReference(win) {
+    if (UTIL.isObject(win)) {
+        windowReference = win;
+    }
+}
+/* start-test-block */
+exports.setWindowReference = setWindowReference;
+/* end-test-block */
+
+function getWindowReference() {
+    return windowReference;
+}
+/* start-test-block */
+exports.getWindowReference = getWindowReference;
+/* end-test-block */
+
+function defineWrapperTargetingKeys(object) {
+    var output = {};
+    UTIL.forEachOnObject(object, function(key, value) {
+        output[value] = "";
+    });
+    return output;
+}
+/* start-test-block */
+exports.defineWrapperTargetingKeys = defineWrapperTargetingKeys;
+/* end-test-block */
+
+function setDisplayFunctionCalledIfRequired(slot, arg) { // TDD, i/o : done
+    /* istanbul ignore else */
+    if (UTIL.isObject(slot) && UTIL.isFunction(slot.getDivID)) {
+        /* istanbul ignore else */
+        if (UTIL.isArray(arg) && arg[0] && arg[0] == slot.getDivID()) {
+            slot.setDisplayFunctionCalled(true);
+            slot.setArguments(arg);
+        }
+    }
+}
+
+/* start-test-block */
+exports.setDisplayFunctionCalledIfRequired = setDisplayFunctionCalledIfRequired;
+/* end-test-block */
+
+function storeInSlotsMap(dmSlotName, currentPhoenixSlot, isDisplayFlow) {
+    // note: here dmSlotName is actually the DivID
+    if (!UTIL.isOwnProperty(refThis.slotsMap, dmSlotName)) {
+        var slot = SLOT.createSlot(dmSlotName);
+        slot.setDivID(dmSlotName);
+        slot.setPubAdServerObject(currentPhoenixSlot);
+        slot.setAdUnitID(currentPhoenixSlot.getAdUnit());
+        slot.setAdUnitIndex(0);
```

Comment:
I think we will need to add adUnitIndex support for UAS as well(in phoenix.js and consume here); can't keep hardcoded 0